### ISSUE TITLE
Compressor power consumption

### ISSUE THREAD

**Issue by jan1472** [pail23/stiebel_eltron_isg_component#342]
Hi,
it would be highly appreciated if the actual compressor power consumption would be available (MODBUS 3679) as an additional sensor. It would make it much easier to handle SG ready automation with respect to the Integration of PV inverters.
Thanks a lot and kind regards,
Jan

**Comment by northalpha** [comment]
This Modbus register is not to be found in any official docs from SE also not in the latest one here: https://www.stiebel-eltron.de/toolbox/content/docs/anleitungen/installation/ISG_Modbus/321798-44755-9770_ISG%20Modbus_de_en_fr_it_nl_cs_sk_pl_hu.pdf

i have tried this for years for my LWZ 5s plus and only came to a really ugly "solution" in combination with HomeAssistant:

1. Have an raspberry pi / something running on the same network
2. Create an API Token for HomeAssistant
3. Have an systemd service `isg2hass.service` on the raspberry pi / something

```
$ systemctl cat isg2hass
# /etc/systemd/system/isg2hass.service
[Unit]
Description=isg2hass
After=network.target

[Service]
ExecStart=/usr/bin/bash /home/isg/isg2hass.sh

[Install]
WantedBy=multi-user.target
```

4. Have an systemd timer `isg2hass.timer`:

```
$ systemctl cat isg2hass.timer
# /etc/systemd/system/isg2hass.timer
[Unit]
Description=isg2hass every minute

[Timer]
OnCalendar=*-*-* *:*:00
Persistent=true
OnBootSec=1min

[Install]
WantedBy=timers.target
```
 
5. The Script (`isg2hass.sh`)

```bash
#!/bin/bash
value_raw=$(curl -s -F 'user=admin' -F 'pass=my.admin.user.password' "my.isg.ip.address/?s=4,7" | grep -A1 MOTORLEISTUNG | grep value | sed -e 's/<[^>]*>//g' | sed 's/[^0-9,]*//g')
value=${value_raw/,/.}
curl -s -X POST \
        -H "Authorization: Bearer "my_token_here \
        -H "Content-Type: application/json" \
        -d '{"state": "'$value'", "attributes": {"unit_of_measurement": "kW", "state_class": "measurement", "device_class": "power"}}' \
         http://my.homeassistant.ip.address:8123/api/states/sensor.my_sensor_name_here
```
(Did you spot the `/?s=4,7`, yes thats an undocumented "hidden" ISG page with even some more stats ...)

5. Profit.
![Image](https://github.com/user-attachments/assets/7f5dc00f-e4d6-4b48-9f19-95c1f306e994)

Maybe, @pail23 is willing / able to implement this "hack" into the integration, but ...

**Comment by Padanian** [comment]
> (Did you spot the `/?s=4,7`, yes thats an undocumented "hidden" ISG page with even some more stats ...)

No, it's not. It's a passive cooling configuration page on my ISG.
Maybe it's firmware-dependent and in that case it can't be implemented.

**Comment by northalpha** [comment]
Interesting, `/?s=4,7` on my ISG Web Firmware `v12.1.2` with activated `EMI` is an `EM-DEBUG-INFOS` page. Maybe the "EMI" or "Energy Mangement" enabled is the difference.
